examples/objectServerExample/src/main/java/io/realm/examples/objectserver/model/CRDTCounter.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2018 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.examples.objectserver.model
+
+import io.realm.MutableRealmInteger
+import io.realm.RealmObject
+import io.realm.annotations.PrimaryKey
+import io.realm.annotations.Required
+
+/**
+ * A named, conflict-free replicated data-type.
+ */
+open class CRDTCounter(@PrimaryKey var name: String): RealmObject() {
+
+    @Required
+    private val counter = MutableRealmInteger.valueOf(0L)
+
+    val count: Long
+        get() = this.counter.get()!!.toLong()
+
+    constructor() : this("") {
```

Comment:
Why not just do:

``` kotlin
open class CRDTCounter: RealmObject() {
    @PrimaryKey
    var name: String

    @Required
    private val counter = MutableRealmInteger.valueOf(0L)
}
```

that way you don't need to juggle a secondary constructor.

Otherwise you could do `@JvmOverloads constructor(@PrimaryKey var name: String = "")` anyways, but I think that is worse.